src/Stratis.Bitcoin.Tests/P2P/PeerConnectorTests.cs

```diff
@@ -210,6 +210,141 @@ public void PeerConnectorDiscovery_ConnectsTo_DiscoveredPeers()
             Assert.Contains(endpointDiscoveredNode, peerConnector.ConnectorPeers.Select(p => p.PeerEndPoint));
         }
 
+        [Fact]
+        public void PeerConnectorDiscovery_ConnectsTo_NodeInSameNetworkGroup_WithIpRangeFilteringDisabled()
+        {
+            Network mainNetwork = Network.Main;
+            DataFolder peerFolder = CreateDataFolder(this);
+            var peerAddressManager = new PeerAddressManager(DateTimeProvider.Default, peerFolder, this.extendedLoggerFactory, new SelfEndpointTracker());
+
+            // IpRangeFiltering enabled by default, disabled explicitly.
+            var nodeSettings = new NodeSettings(mainNetwork, args: new[] { "-IpRangeFiltering=false" });
+            
+            var connectionManagerSettingsExisting = new ConnectionManagerSettings(nodeSettings);
+            Mock<INetworkPeerFactory> networkPeerFactoryExisting = new Mock<INetworkPeerFactory>();
+
+            var peerConnector = new PeerConnectorConnectNode(this.asyncLoopFactory, DateTimeProvider.Default, this.extendedLoggerFactory, this.network, networkPeerFactoryExisting.Object, this.nodeLifetime, nodeSettings, connectionManagerSettingsExisting, peerAddressManager, new SelfEndpointTracker());
+
+            IConnectionManager connectionManagerExisting = this.CreateConnectionManager(nodeSettings, connectionManagerSettingsExisting, peerAddressManager, peerConnector);
+            peerConnector.Initialize(connectionManagerExisting);
+            
+            // Peer 1.
+            IPAddress originalAddressPeer1 = IPAddress.Parse("::ffff:57.48.183.81"); // ipv4
+            var endpointPeer1 = new IPEndPoint(originalAddressPeer1, 80);
+            peerAddressManager.AddPeer(endpointPeer1, IPAddress.Loopback);
+            bool connectedToThisPeer = ConnectToPeer(peerAddressManager, networkPeerFactoryExisting, connectionManagerSettingsExisting, peerConnector, endpointPeer1);
+            Assert.True(connectedToThisPeer);
+
+            // Peer 2 has different network group.
+            IPAddress addressInDifferentNetworkGroupPeer2 = IPAddress.Parse("99be:f5c5:adc2:525c:f6d7:7b30:5336:5a0f");  // ipv6
+            var endpointPeer2 = new IPEndPoint(addressInDifferentNetworkGroupPeer2, 80);
+            peerAddressManager.AddPeer(endpointPeer2, IPAddress.Loopback);
+            connectedToThisPeer = ConnectToPeer(peerAddressManager, networkPeerFactoryExisting, connectionManagerSettingsExisting, peerConnector, endpointPeer2);
+            Assert.True(connectedToThisPeer);  // Different network group: connects.
+
+            // Peer 3 in same network group as Peer 2.
+            IPAddress addressInSameNetworkGroupPeer3 = IPAddress.Parse("99be:f5c5:adc2:525c:db45:d36e:ce01:a394"); // ipv6
```

Comment:
Use these exact two peers also in the test range filtering is enabled

Reply:
As above ^^^